arch/arc/core/prep_c.c

```diff
@@ -121,6 +122,10 @@ extern FUNC_NORETURN void z_cstart(void);
 
 void z_prep_c(void)
 {
+#if defined(CONFIG_PLATFORM_PREP_HOOK)
```

Comment:
would it make sense to have weak functions instead of config options? it's already the case for other arch functions (`arch_system_halt`, `arch_spin_relax`...) would save few options, it's not like you may ever want to implement it and turn the option off anyway

Reply:
I think having weak defaults makes sense.

Reply:
the `__weak` approach is mentioned in the RFC as an option. We had many issues with weak functions in the past (defined at multiple levels (arch, soc, board) so you never knew which one you got etc, maybe it is not the case here. I also remember some compilers/architectures having problems with weak symbols the reason why existing hooks did not use them. Anyways, lets see if there is more feedback input to that and we can switch to weak definitions.

Reply:
this way adds a lot of #ifdef, __weak would be nice yes (if there aren't any issues with it)

Reply:
at least for one of the hooks and for some platforms, the hooks is being disabled in some samples...

```
samples/drivers/ipm/ipm_mcux/remote/prj.conf:CONFIG_PLATFORM_RESET_HOOK=n
samples/subsys/ipc/openamp/remote/prj.conf:CONFIG_PLATFORM_RESET_HOOK=n
samples/subsys/ipc/openamp_rsc_table/boards/colibri_imx7d_mcimx7d_m4.conf:CONFIG_PLATFORM_RESET_HOOK=n
samples/subsys/ipc/openamp_rsc_table/boards/stm32mp157c_dk2.conf:CONFIG_PLATFORM_RESET_HOOK=n
```

need to find a out why and figure a better way to deal with this using weak symbols...